packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -266,6 +316,36 @@ class TextPainter {
 
   ui.Paragraph _layoutTemplate;
 
+  /// An ordered list of [TextBox]es that bound the positions of the placeholders
+  /// in the paragraph.
+  ///
+  /// Each box corresponds to a [PlaceholderSpan] in the order they were defined
+  /// in the [InlineSpan] tree.
+  List<TextBox> get inlinePlaceholderBoxes => _inlinePlaceholderBoxes;
+  List<TextBox> _inlinePlaceholderBoxes;
+
+  /// An ordered list of scales for each placeholder in the paragraph.
+  ///
+  /// Each scale corresponds to a [PlaceholderSpan] in the order they were defined
+  /// in the [InlineSpan] tree.
+  List<double> get inlinePlaceholderScales => _inlinePlaceholderScales;
+  List<double> _inlinePlaceholderScales;
+
+  /// Sets the dimensions of each placeholder in [text].
+  /// 
+  /// The number of [PlaceholderDimensions] provided should be the same as the
```

Comment:
Can we add an assert for this? (Or are the PlaceholderSpans at this point not known yet?)